napi.h

```diff
@@ -2242,8 +2249,55 @@ namespace Napi {
     napi_threadsafe_function _tsfn;
   };
 
+  template <typename DataType>
+  class AsyncProgressWorkerBase : public AsyncWorker {
+    public:
+     virtual void OnWorkProgress(DataType* data) = 0;
+     class ThreadSafeData {
+       public:
+        ThreadSafeData(AsyncProgressWorkerBase* asyncprogressworker, DataType* data)
+          : _asyncprogressworker(asyncprogressworker), _data(data) {}
+
+        AsyncProgressWorkerBase* asyncprogressworker() { return _asyncprogressworker; };
```

Comment:
These methods are very simple and naïve field exposing. I'm noticed there are previous works that also inlined these methods for readability. Since there are no acutal logics in the block, I'd believe it's ok to put them here?